index.js

```diff
@@ -34,23 +34,31 @@ function fastifyAccepts (fastify, options, done) {
   fastify.decorateRequest('accepts', acceptsMethod)
 
   methodNames.forEach(methodName => {
-    fastify.decorateRequest(methodName, function (arr) {
-      const acceptsObject = this.accepts()
-      if (arguments.length === 0) return acceptsObject[methodName]()
-      return acceptsObject[methodName](arr)
-    })
+    // Defining methods this way to ensure named functions show in stack traces
+    fastify.decorateRequest(methodName, {
+      [methodName]: function (arr) {
+        const acceptsObject = this.accepts()
+        if (arguments.length === 0) return acceptsObject[methodName]()
+        return acceptsObject[methodName](arr)
+      }
+    }[methodName])
   })
 
   if (options.decorateReply === true) {
     fastify.decorateReply('requestAccepts', replyAcceptMethod)
 
     methodNames.forEach(methodName => {
       const capitalizedMethodName = methodName.replace(/(?:^|\s)\S/gu, a => a.toUpperCase())
-      fastify.decorateReply('request' + capitalizedMethodName, function (arr) {
-        const acceptsObject = this.requestAccepts()
-        if (arguments.length === 0) return acceptsObject[methodName]()
-        return acceptsObject[methodName](arr)
-      })
+      const replyMethodName = 'request' + capitalizedMethodName
+      const acceptsMethodName = 'accepts' + capitalizedMethodName
+      // Defining methods this way to ensure named functions show in stack traces
+      fastify.decorateReply(replyMethodName, {
+        [acceptsMethodName]: function (arr) {
+          const acceptsObject = this.requestAccepts()
+          if (arguments.length === 0) return acceptsObject[methodName]()
+          return acceptsObject[methodName](arr)
+        }
+      }[acceptsMethodName])
```

Comment:
So, with this trick we get the `acceptsMethodName` in the debugger, instead of a generic function name in the case we:

```js
astify.decorateReply(replyMethodName, function foo(arr) {}
```

Interesting!